doc/docusaurus/docs/working-with-scripts/script-purposes.md

```diff
@@ -0,0 +1,80 @@
+Next we go over and explain all these different script purposes.
+
+## Spending
+
+A spending script validates the spending of a UTXO.
+A UTXO can have either a public key address or a script address.
+To spend a UTXO with a script address, the script whose hash matches the script address must be included in the transaction (either directly, or as a reference script), and is executed to validate it.
+
+The script can make the decision based on the datum attached in the UTXO being spent, if any (datum is mandatory for Plutus V1 and V2, i.e., a UTXO with a Plutus V1 or V2 script address but without a datum cannot be spent; datum is optional for Plutus V3), the redeemer included in the transaction, as well as other fields of the transaction.
+
+The `Spending` constructor of `ScriptPurpose` contains a `TxOutRef` field, which is the UTXO the transaction is attempting to spend.
+
+## Minting
+
+Minting scripts, sometimes also referred to as minting policies, are used to approve or reject minting of new assets.
+
+An asset on Cardano contains a `CurrencySymbol` and a `TokenName`.
+If a transaction attempts to mint some new assets, then for each unique `CurrencySymbol` in the new assets, the script whose hash matches the `CurrencySymbol` must be included in the transaction, and is executed to validate the minting of that `CurrencySymbol`.
+This is the reason why the `Minting` constructor of `ScriptPurpose` contains a `CurrencySymbol` field.
+
+A minting script should pay attention to the fact that the transaction may attempt to mint multiple assets with the same `CurrencySymbol` but different `TokenNames`.
+In general all these assets with the same `CurrencySymbol` should be checked.
+
+## Rewarding
+
+As previously stated, a UTXO's address can be either a public key address or a script address.
+Both kinds of addresses can optionally have a staking credential.
+A UTXO may contain Ada, and the Ada it contains can be delegated to an SPO to earn staking rewards.
+Staking rewards are deposited into a reward account[^1] corresponding to the staking credential.
+
+A staking credential can contain either a public key hash or a script hash.
+To withdraw rewards from a reward account corresponding to a staking credential that contains a script hash, the script with that particular hash must be included in the transaction, and is executed to validate the withdrawal.
+
+For example, such a script may check that any withdrawal must not take more than 10 Ada at a time, and that any withdrawal must be deposited into one of the allowed addresses.
+
+## Certifying
+
+A certifying script validates one of the following actions a transaction attempts to make: (1) registering a staking credential, and in doing so, creating a reward account associated with the staking credential; (2) de-registering a staking credential, and in doing so, terminating the reward account; (3) delegating a staking credential to a particular delegatee.
```

Comment:
There are more actions related to certificates (see [haddock](https://plutus.cardano.intersectmbo.org/haddock/latest/plutus-ledger-api/PlutusLedgerApi-V3.html#t:TxCert)). But maybe it'll be too much to go over each one. We could do something like:

"A certifying script can validate many certificate-related transactions. For example:"

That suggests there are more certificates without needing a significant rewrite of this section.